wic/src/main/java/edu/asu/diging/wic/core/conceptpower/impl/ConceptMapper.java

```diff
@@ -85,4 +85,61 @@ public IConcept mapConceptpowerConceptToConcept(ConceptpowerConcept conceptpower
         
         return concept;
     }
+    
+    /* (non-Javadoc)
+     * @see edu.asu.diging.wic.core.conceptpower.IConceptMapper#mapConceptpowerConceptToConcept(edu.asu.diging.wic.core.conceptpower.impl.ConceptEntry)
+     */
+    @Override
+    public IConcept mapConceptpowerConceptToConcept(ConceptEntry conceptEntry) {
+        IConcept concept = new Concept();
+        IConceptType conceptType = conceptEntry.getType();
+
+        concept.setId(conceptEntry.getConeptEntryId().getConceptId());
+        concept.setUri(conceptEntry.getConeptEntryId().getConceptUri());
+
+        concept.setWord(conceptEntry.getLemma());
+
+        concept.setPos(conceptEntry.getPos());
+
+        concept.setDescription(conceptEntry.getDescription());
+
+        concept.setConceptList(conceptEntry.getConceptList());
+
+        concept.setTypeId(conceptType != null && conceptType.getUri() != null ? conceptType.getUri() : "");
+
+        List<ConceptEntryId> altIds = conceptEntry.getAlternativeIds();
+        List<String> ids = new ArrayList<>();
+        altIds.forEach(id -> ids.add(id.getConceptUri()));
+        concept.setAlternativeUris(ids);
+
+        concept.setCreatorId(conceptEntry.getCreatorId());
+
+        if (conceptEntry.getEqualTo() != null) {
+            concept.setEqualTo(conceptEntry.getEqualTo());
+        } else {
+            concept.setEqualTo(new ArrayList<>());
+        }
+
+        if (conceptEntry.getSimilarTo() != null) {
+            concept.setSimilarTo(conceptEntry.getSimilarTo());
+        } else {
+            concept.setSimilarTo(new ArrayList<>());
+        }
+
+        if (conceptEntry.getWordnetIds() != null) {
+            concept.setWordnetIds(conceptEntry.getWordnetIds());
+        } else {
+            concept.setWordnetIds(new ArrayList<>());
+        }
+
+        if (conceptType != null) {
+            IConceptType type = new ConceptType();
+            type.setUri(conceptType.getUri());
+            type.setId(conceptType.getId());
+            type.setDescription("");
+            type.setName(conceptType.getName());
+            concept.setType(type);
+        }
+        return concept;
```

Comment:
The code above is pretty much exactly the same as for the other `mapConceptpowerConceptToConcept` method, isn't it? There should be no code duplication. Instead have both classes (`ConceptEntry` and `ConceptpowerConcept`) implement the same interface and change the method signature to accept that interface.